### DISCUSSION TITLE
Support for Financial-grade API (FAPI)?

### DISCUSSION THREAD

**Discussion by kmukku** [nextauthjs/next-auth#6049] [Help]
I have a use case where I as a Relying Party (RP) need to provide keys via JWKS-uri to OpenID Connect Provider (OP). 

I'm not really familiar with this approach, but it seems to be related to Financial-grade API (FAPI) requirements and especially to use of asymmetrical key pairs and to the exchange of JWT.

The spec for FAPI can be found [from here](https://openid.net/specs/openid-financial-api-part-2-1_0.html) and 8.9. states:

> This profile requires both Clients and Authorization Servers to verify payloads with keys from the other party. 

Which sounds pretty much I'm trying to achieve. I tried to look documentation but couldn't find any hints how to provide keys. This issue #3150 might be somewhat related. 

Anyone has any insight on this?

**Comment by kmukku** [comment]
I dug into the topic a little more and it seems that at least the [node-openid-client](https://github.com/panva/node-openid-client) is [FAPI certified](https://openid.net/developers/certified/#FAPIRP).

**Comment by alvinangjh** [comment]
Just to hop onto this same topic, may I know if FAPI 2.0 is supported with next-auth? I tried to search information around, but I can't seem to find anything on this topic.
